src/textual/debounce.py

```diff
@@ -0,0 +1,24 @@
+import asyncio
+from functools import wraps
+from typing import Callable, Optional
+
+from .app import App
+from .timer import Timer as TimerClass
+
+
+def debounce(timeout: float) -> Callable:
```

Comment:
Can we have a docstring for this? In the google format. Should be plenty of examples in the repo.